### PR TITLE
Feature/internal relations

### LINKED ISSUE #28: Relations to other diagnostic arguments
A field prepopulated with annotation-ids (labeled with argument name) from the Zotero group library, and an option to specify the type of relation (confirms, contradicts, etc.; Jan will provide a list).

### PR DESCRIPTION
**Pull request by tijmenbaarda**
Add a form field to refer to another argument in the current library. The list is obtained in `relations.js` by scanning the Zotero library belonging to the selected annotation for Lidia annotations.

* PDFs without a parent item are not included in the list. I think this is fine because supporting PDFs without a parent item brings up additional problems (they have no bibliographic information attached)
* The Zotero keys are used as identifier. Are these guaranteed to be unique and persistent? (I think they are at least unique when working in a synchronized group library. We need to test if they remain the same when moving a PDF from a user library into a group library.)
* Only one relation can be defined. Do we need to support multiple relations?
* No support for reverse relations, although it would not be difficult to implement showing these.
* Do we need special support for symmetric relations?

Related to scalability:
* If there are many annotations, the combobox becomes too full
* The complete list of annotations is scanned every time the user selects an annotation, to guarantee that the user always sees all existing annotations. This may become too slow when there is a large library

We may solve these last two issues perhaps by providing a separate dialogue window for selection (with search functionality), but I would suggest leaving this for now so that the users can test.

Closes #28 (at least partially)

### CODE REVIEW

File: src/panel.js [hunk 1]
```diff
@@ -26,6 +27,7 @@ export class LidiaPanel {
      */
     async buildSideBarPanel() {
         log("Building LIDIA panel");
+        await getAllLidiaAnnotations();
```
Comment (ar-jan) [inline src/panel.js:30]: This is done (and stored) in `receiveAnnotation()`, is this needed here?

File: src/relations.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    if (!libraryID) {
+        libraryID = Zotero.Libraries.userLibraryID;
+    }
+    /* Find all items in the requested library; get their attached PDFs and
+     * in turn all their annotations. Check for each annotation if it is a
+     * LIDIA annotation and return a list of all of these
+     */
+    const s = new Zotero.Search();
+    s.libraryID = libraryID;
+    s.addCondition('recursive', 'true');
```
Comment (ar-jan) [inline src/relations.js:20]: I wonder if we could use `s.addCondition('itemType', 'is', 'annotation');` to get all annotations (not tested), and then get the `parentItem` to get the publication title and key per annotation. If it's possible, it should be faster on large libraries and require less nesting.

**Review by ar-jan — COMMENTED** [review]
Looks good, worth checking whether something like `s.addCondition('itemType', 'is', 'annotation');` would work here.

**Comment by ar-jan** [comment]
> Do we need special support for symmetric relations?

I think we only have directional relationships (currently)? That also simplifies managing the relationships, since we don't have the benefit of referential integrity via a database. Maybe we should add an option to check whether all relationships are still valid.

File: src/panel.js (hunk 1 above)
Comment (tijmenbaarda) [reply src/panel.js:30]: No, this was a leftover.

File: src/relations.js (hunk 2 above)
Comment (tijmenbaarda) [reply src/relations.js:20]: This does not work -- apparently this does not give back the annotations but a list of the PDF items that belong to them (I have not looked into that in detail).

**Pull request merged**